Desktop/TypescriptExercises/day2 - angular form exercise/src/app/models/app.logic.ts

```diff
@@ -21,4 +21,42 @@ export class Logic {
       this.products.push(prd);
       return this.products;
     }
-}
+    searchProducts(searchText: string): Array<Product>
+    {
+      let searchResults: Array<Product> = []
+      this.products.forEach(element => {
+        if(element.Category.toLowerCase().includes(searchText.toLowerCase()) || element.ProductName.toLowerCase().includes(searchText.toLowerCase()))
+          {
+            searchResults.push(element)
+          }
+      });
+      return searchResults
+    }
+    fetchIndex(prd:Product): number
+    {
+      return this.products.map(function(e) { return e.ProductId }).indexOf(prd.ProductId);
+    }
+    updateProduct(prd:Product): Array<Product>{
+      let indexToUpdate = this.fetchIndex(prd)
+      if(indexToUpdate == -1)
+        {
+          alert("Invalid product Id")
+          return this.products
+        }
+      else
+      {
+        this.products[indexToUpdate].Category = prd.Category
```

Comment:
You can use splice(index, 1, updatedProduct);

Reply:
but how would I get the index?

Reply:
oh alright, got it

Reply:
done